session/txn.go

```diff
@@ -282,6 +289,11 @@ func (txn *LazyTxn) changePendingToValid(ctx context.Context) error {
 	txn.Transaction = t
 	txn.initStmtBuf()
 
+	if txn.enterAggressiveLockingOnValid {
+		txn.enterAggressiveLockingOnValid = false
+		txn.Transaction.StartAggressiveLocking()
```

Comment:
We may need telemetry information for example how many transactions use the aggressive locking mode and actually do the aggressive-mode locking. As this is a new feature we could enrich this statistic information early.